slicer/index.js

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env node
+
+const debug = require('debug')('slicer');
+const fs = require('fs');
+const co = require('co');
+const dotenv = require('dotenv').config();
+const uuid = require('uuid').v4;
+
+const imageProcessing = require('./lib/image-processing');
+
+const database = require('./lib/database-interface');
+
+const databaseConnectionDetails = {
+	host : process.env.SLICE_DATABASE_HOST,
+	user : process.env.SLICE_DATABASE_USERNAME,
+	password : process.env.SLICE_DATABASE_PASSWORD,
+	port : process.env.SLICE_DATABASE_PORT
+};
+
+const tmpPath = process.env.TMPPATH || '/tmp/';
+
+const AWS = require('aws-sdk');
+const S3 = new AWS.S3();
+
+let currentJob;
+let isProcessing = false;
+
+function getJobFromSliceQueue(){
+	// Get item from queue and return it
+	return database.query('SELECT * FROM slice ORDER BY rand() LIMIT 1')
+		.then(function(data){
+			return data.results[0];
+		})
+		.then(function(job){
+			return database.query(`UPDATE slice SET status = 'processing' WHERE id=${job.id}`)
+				.then(function(){
+					return job;
+				})
+		})
+	;
+}
+
+function deleteJobFromSliceQueue(id){
+	// Delete job from slice queue when it's been completed
+	return database.query(`DELETE * FROM slice WHERE id = ${id}`);
```

Comment:
check the status value before deleting

Reply:
The job is only deleted when it's been successfully completed. Would the check for the status be a little redundant?